### PR TITLE
 Add support for SHA256, SHA512 and Blowfish

### PR DESCRIPTION
**Pull request by dtitov**
Also:
- add support for DSS authentication keys;
- add healthcheck-endpoint;
- add Travis Docker auto-push on successful PR merge.

### CODE REVIEW

**Comment by dtitov** [comment]
Coveralls thinks that these lines are not covered: https://coveralls.io/builds/18195900/source?filename=src%2Fmain%2Fjava%2Fse%2Fnbis%2Flega%2Finbox%2Fsftp%2FInboxAuthenticator.java 

But that's not true, they are covered in the `se.nbis.lega.inbox.sftp.InboxAuthenticatorTest#authenticatePasswordFail` test.

So would ignore Coveralls panic about coverage going down...

File: pom.xml [hunk 1]
```diff
@@ -105,8 +113,13 @@
                 <configuration>
                     <imageName>nbisweden/ega-mina-inbox</imageName>
                     <baseImage>openjdk:8-jre-alpine</baseImage>
+                    <!-- Next line needs to be a single line, otherwise plugin crashes-->
                     <entryPoint>["java", "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005", "-jar", "/${project.build.finalName}.jar", "-XX:+UnlockExperimentalVMOptions", "-XX:+UseCGroupMemoryLimitForHeap"]
                     </entryPoint>
+                    <healthcheck>
+                        <options>--interval=10s</options>
+                        <cmd>curl --fail http://localhost:8080/actuator/health || exit 1</cmd>
```
Comment (blankdots) [inline pom.xml]: I think we can modify the `health` endpoint address, maybe make it simpler (not using the default one):
```
management.endpoints.web.base-path=/
management.endpoints.web.path-mapping.health=health
```
https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html#production-ready-endpoints-custom-mapping

**Comment by blankdots** [comment]
It would be good to have an unit test for this:
https://github.com/NBISweden/LocalEGA-inbox/blob/feature/blowfish/src/main/java/se/nbis/lega/inbox/sftp/CredentialsProvider.java#L46-L48

and this:
https://github.com/NBISweden/LocalEGA-inbox/blob/feature/blowfish/src/main/java/se/nbis/lega/inbox/sftp/InboxAuthenticator.java#L124

File: pom.xml (hunk 1 above)
Comment (dtitov) [reply pom.xml]: Done.

**Comment by dtitov** [comment]
I've added some more tests.

**Pull request merged**